### PR TITLE
Magazines

### PR DESCRIPTION
**Pull request by solocommand**
Adds support for magazine list, issue list, and issue content pages

# /magazine
screencapture-0-0-0-0-10007-magazine-2019-04-04-10_21_16
# /magazine/publicationId
screencapture-0-0-0-0-10007-magazine-53f6823928ed57018875bb53-2019-04-04-10_21_05
# /magazine/issueId
screencapture-0-0-0-0-10007-magazine-47305-2019-04-04-10_20_41

### CODE REVIEW

File: services/marko-example-website/server/templates/magazine/components/layout.marko [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { getAsObject } from '@base-cms/object-path';
+
+$ const publications = getAsObject(input, 'publications');
+
+<document>
+  <@head>
+    <cms-page-metadata for="magazine" ...publications />
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/components/layout.marko]: Make a distinct layout template per type. Update `cms-page-metadata for "magazine-publication", etc. Also, manually add title and meta description for the root publication index using `cms-page-title` and `cms-page-meta-description`

File: services/marko-example-website/server/templates/magazine/index.marko [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <div class="card-body pb-0">
+
+          <nav aria-label="breadcrumb">
+            <ol class="breadcrumb">
+              <li class="text-uppercase breadcrumb-item small">Magazines</li>
+            </ol>
+          </nav>
+
+          <h1 class="mb-2">Officer.com Magazines</h1>
+          <p>Law Enforcement Technology (LET) and Law Enforcement Product News (LEPN) are the leading magazine publications for the police and law enforcement community.</p>
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/index.marko]: Add this to site config file

File: services/marko-example-website/server/templates/magazine/issue.marko [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  <div class="row">
+    <div class="col">
+      <div class="card">
+        <div class="card-body pb-0">
+
+          <nav aria-label="breadcrumb">
+            <ol class="breadcrumb">
+              <li class="text-uppercase breadcrumb-item small"><a href="/magazine">Magazines</a></li>
+              <li class="text-uppercase breadcrumb-item small"><a href=`/magazine/${issue.publication.id}`>${issue.publication.name}</a></li>
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/issue.marko]: This should use canonicalPath with cms-website-link

File: services/marko-example-website/server/templates/magazine/issue.marko [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  </div>
+
+  <div class="row">
+    <div class="col-6 col-md-3">
+      <cms-magazine-issue-link obj=issue class="col-6">
+        <cms-magazine-issue-image obj=issue.coverImage alt=issue.coverDescription />
+      </cms-magazine-issue-link>
+    </div>
+    <div class="col-6 col-md-6">
+      <cms-text-element obj=issue field='description' class="lead" />
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/issue.marko]: Also, always use `"` quotes for marko props

File: services/marko-example-website/server/templates/magazine/publication.marko [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  <cms-ad-gam-display id="gpt-ad-bs" class="text-center" />
+
+  <div class="row">
+    <div class="col">
+      <div class="card">
+        <div class="card-body pb-0">
+
+          <nav aria-label="breadcrumb">
+            <ol class="breadcrumb">
+              <li class="text-uppercase breadcrumb-item small"><a href="/magazine">Magazines</a></li>
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/publication.marko]: canonicalPath here

File: services/marko-example-website/server/templates/magazine/issue.marko [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  <cms-ad-gam-display id="gpt-ad-bs" class="text-center" />
+
+  <div class="row">
+    <div class="col">
+      <div class="card">
+        <div class="card-body pb-0">
+
+          <nav aria-label="breadcrumb">
+            <ol class="breadcrumb">
+              <li class="text-uppercase breadcrumb-item small"><a href="/magazine">Magazines</a></li>
```
Comment (zarathustra323) [inline services/marko-example-website/server/templates/magazine/issue.marko]: `out.global.req.path`

**Comment by zarathustra323** [comment]
@solocommand Check all queries and add to the generate index script in the root of graphql-server

**Comment by solocommand** [comment]
@solocommand All up to date

**Comment by solocommand** [comment]
@zarathustra323 all up to date!

**Pull request merged**